### PR TITLE
Update all dependencies 

### PR DESCRIPTION
**Pull request by joshuawilson**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,135 +68,108 @@
...
-    "@types/core-js": "0.9.43",
-    "@types/jasmine": "2.6.2",
-    "@types/lodash": "4.14.80",
-    "angular2-load-children-loader": "0.2.1",
-    "angular2-router-loader": "0.3.5",
+    "@ngtools/webpack": "6.0.8",
+    "@types/core-js": "2.5.0",
+    "@types/jasmine": "2.8.8",
+    "@types/lodash": "4.14.112",
+    "ajv": "6.5.2",
```
Comment (christianvogt) [inline package.json]: Why is this needed?

File: package.json [hunk 2]
```diff
@@ -68,135 +68,108 @@
...
     "angular2-template-loader": "0.6.2",
-    "assets-webpack-plugin": "3.5.1",
-    "autoprefixer": "7.1.6",
-    "awesome-typescript-loader": "3.2.3",
+    "autoprefixer": "8.6.5",
```
Comment (christianvogt) [inline package.json]: Why is this needed? I thought there's no css.

File: src/app/auth/authentication.service.ts [hunk 3]
```diff
@@ -135,27 +142,31 @@ export class AuthenticationService {
...
+        .pipe(
+          map((t: any) => {
+            let token = this.processTokenResponse(t.token);
+            this.clearTimeoutId = null;
+            this.setupRefreshTimer(token.expires_in);
+            return token;
+          }),
+          catchError(response => {
+            // Additionally catch a 400 from keycloak
+            if (response.status === 400 || isAuthenticationError(response)) {
```
Comment (christianvogt) [inline src/app/auth/authentication.service.ts]: I don't think the call to`isAuthenticationError` needed here still if we have the interceptor.

File: src/app/user/user.service.ts [hunk 4]
```diff
@@ -60,33 +63,33 @@ export class UserService {
     this.searchUrl = apiUrl + 'search';
     // TODO - switch to internal observable that is populated on initialization
     // and only expose currentLoggedInUser publicly
-    this.loggedInUser = Observable.merge(
+    this.loggedInUser = merge(
       broadcaster.on('loggedin')
-        .map(val => 'loggedIn'),
+        .pipe(map((val: any) => 'loggedIn')),
```
Comment (christianvogt) [inline src/app/user/user.service.ts]: Can you not simply remove val?
`.pipe(map(() => 'loggedIn'))`

File: package.json (hunk 1 above)
Comment (joshuawilson) [reply package.json]: It fixed a warning but now when I remove it there is no warning, hmm.
I'll take it out for now.

File: package.json (hunk 2 above)
Comment (joshuawilson) [reply package.json]: There was a dependency the first time I tried to remove it. I later removed it but not autoprefixer, good catch.

File: src/app/auth/authentication.service.ts (hunk 3 above)
Comment (joshuawilson) [reply src/app/auth/authentication.service.ts]: I'm unsure about this one. It triggers the broadcast if it is there. As that was added to deal with something, I would almost rather leave it alone and refactor after.

File: src/app/user/user.service.ts (hunk 4 above)
Comment (joshuawilson) [reply src/app/user/user.service.ts]: good point

**Comment by joshuawilson** [comment]
updated for webpack 4 and removed unused webpack configs

File: package.json [hunk 5]
```diff
@@ -68,135 +66,107 @@
...
-    "@angular/common": "^4.4.6",
-    "@angular/compiler": "^4.4.6",
-    "@angular/compiler-cli": "^4.4.6",
-    "@angular/core": "^4.4.6",
-    "@angular/forms": "^4.4.6",
-    "@angular/http": "^4.4.6",
-    "@angular/platform-browser": "^4.4.6",
-    "@angular/platform-browser-dynamic": "^4.4.6",
-    "@angular/router": "^4.4.6",
+    "@angular/common": "6.1.0-rc.0",
```
Comment (corinnekrych) [inline package.json]: shall we so to `^6.0.9` instead of fixing version in the lib ?

File: src/app/auth/authentication.service.ts (hunk 3 above)
Comment (rohitkrai03) [reply src/app/auth/authentication.service.ts]: I think we can remove these explicit calls to `isAuthenticationError` since we have interceptors for this in place now.

File: src/app/user/user.service.ts [hunk 6]
```diff
@@ -60,33 +64,33 @@ export class UserService {
...
+        .pipe(map(() => 'authenticationError'))
+    ).pipe(
+      switchMap((val: any) => {
         // If it's a login event, then we need to retreive the user's details
         if (val === 'loggedIn') {
-          return this.http
-            .get(this.userUrl, { headers: this.headers })
-            .map(response => cloneDeep(response.json().data as User));
+          return this.http.get(this.userUrl, { headers: this.headers })
+            .pipe(map(response => cloneDeep(response.json().data as User)));
```
Comment (corinnekrych) [inline src/app/user/user.service.ts]: let's move to `HttpClient` and remove the need of `json()`

File: src/app/user/user.service.ts [hunk 7]
```diff
@@ -97,25 +101,27 @@ export class UserService {
   getUserByUserId(userId: string): Observable<User> {
     return this.http
       .get(`${this.usersUrl}/${encodeURIComponent(userId)}`, { headers: this.headers })
-      .map(response => {
-        return response.json().data as User;
-      })
-      .catch(this.catchRequestError);
+      .pipe(
+        map(response => {
+          return response.json().data as User;
```
Comment (corinnekrych) [inline src/app/user/user.service.ts]: let's move to `HttpClient` and remove the need of json()

File: src/app/user/user.service.ts [hunk 8]
```diff
@@ -125,12 +131,14 @@ export class UserService {
     if (search && search !== '') {
       return this.http
         .get(this.searchUrl + '/users?q=' + encodeURIComponent(search), {headers: this.headers})
-        .map(response => {
-          return response.json().data as User[];
-        })
-        .catch(this.catchRequestError);
+        .pipe(
+          map(response => {
+            return response.json().data as User[];
```
Comment (corinnekrych) [inline src/app/user/user.service.ts]: idem. 
let's move to `HttpClient` and remove the need of `json()`

File: src/app/user/user.service.ts [hunk 9]
```diff
@@ -184,10 +194,12 @@ export class UserService {
   filterUsersByUsername(username: string): Observable<User[]> {
     return this.http
       .get( `${this.usersUrl}?filter[username]=${encodeURIComponent(username)}`, { headers: this.headers })
-      .map(response => {
-        return response.json().data as User[];
-      })
-      .catch(this.catchRequestError);
+      .pipe(
+        map(response => {
+          return response.json().data as User[];
```
Comment (corinnekrych) [inline src/app/user/user.service.ts]: idem. 
let's move to `HttpClient` and remove the need of `json()`

**Comment by corinnekrych** [comment]
@joshuawilson here is my review: https://github.com/joshuawilson/ngx-login-client/pull/2 and work on using `HttpClient` everywhere

**Comment by joshuawilson** [comment]
[test]
